src/test/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunctionTest.java

```diff
@@ -1686,4 +1690,89 @@ public void testInvalidUseExtensionLabel() throws Exception {
             + " '_', '.' and '+'");
   }
+
+  @Test
+  public void testSingleVersionOverridePatches() throws Exception {
+    FakeRegistry registry = registryFactory.newFakeRegistry("/foo");
+    ModuleFileFunction.REGISTRIES.set(differencer, ImmutableSet.of(registry.getUrl()));
+    ModuleKey bbb = createModuleKey("bbb", "1.0");
+    registry.addModule(bbb, "module(name='bbb',version='1.0')");
+
+    scratch.file("BUILD");
+    scratch.file(
+        "patch.diff",
+        """
+        diff --git a/MODULE.bazel b/MODULE.bazel
+        --- a/MODULE.bazel
+        +++ b/MODULE.bazel
+        @@ -1,1 +1,1 @@
+        -module(name='bbb',version='1.0')
+        +module(name='bbb',version='1.0',bazel_compatibility=[">=7.0.0"])
+        diff --git a/not/MODULE.bazel b/not/MODULE.bazel
+        --- a/not/MODULE.bazel
+        +++ b/not/MODULE.bazel
+        @@ -1,1 +1,2 @@
+         module(name='bbb',version='1.0',bazel_compatibility=[">=7.0.0"])
+        +bazel_dep(name='ccc',version='3.0')
+        """);
+    scratch.file("other/pkg/BUILD");
+    Path otherPatch =
+        scratch.file(
+            "other/pkg/other_patch.diff",
+            """
+        --- a/MODULE.bazel
+        +++ b/MODULE.bazel
+        @@ -1,1 +1,2 @@
+         module(name='bbb',version='1.0',bazel_compatibility=[">=7.0.0"])
+        +bazel_dep(name='ccc',version='3.0')
+        """);
+
+    var moduleFileKey =
+        ModuleFileValue.key(
+            bbb,
+            SingleVersionOverride.create(
+                Version.EMPTY,
+                "",
+                ImmutableList.of(
+                    Label.parseCanonicalUnchecked("//:patch.diff"),
+                    Label.parseCanonicalUnchecked("@other_repo//:patch.diff"),
+                    Label.parseCanonicalUnchecked("//other/pkg:other_patch.diff")),
+                ImmutableList.of(),
+                1));
+    EvaluationResult<ModuleFileValue> result =
+        evaluator.evaluate(ImmutableList.of(moduleFileKey), evaluationContext);
+    if (result.hasError()) {
+      throw result.getError().getException();
+    }
+    assertThat(result.get(moduleFileKey).getModule().getBazelCompatibility())
+        .isEqualTo(ImmutableList.of(">=7.0.0"));
+    assertThat(result.get(moduleFileKey).getModule().getDeps())
+        .containsExactly(
+            "ccc",
+            InterimModule.DepSpec.fromModuleKey(ModuleKey.create("ccc", Version.parse("3.0"))));
+
+    FileSystemUtils.writeContentAsLatin1(
+        otherPatch,
+        """
+        --- a/MODULE.bazel
+        +++ b/MODULE.bazel
+        @@ -1,1 +1,3 @@
+         module(name='bbb',version='1.0',bazel_compatibility=[">=7.0.0"])
+        +bazel_dep(name='ccc',version='2.0')
+        """);
+    differencer.invalidate(
+        ImmutableList.of(
+            FileValue.key(RootedPath.toRootedPath(Root.fromPath(rootDirectory), otherPatch))));
+
+    result = evaluator.evaluate(ImmutableList.of(moduleFileKey), evaluationContext);
```

Comment:
@Wyverald This call returns the stale value and I don't understand why. I can tell that the `difference.invalidate` call actually marks the correct nodes for invalidation, but somehow `ModuleFileFunction` isn't rerun even though it has a dep registered.

Do you have any idea how I could debug this further?

Reply:
Nvm, fixed